terraform/lib/dependabot/terraform/registry_client.rb

```diff
@@ -98,6 +81,47 @@ def headers_for(hostname)
         token = tokens[hostname]
         token ? { "Authorization" => "Bearer #{token}" } : {}
       end
+
+      def services
+        @services ||=
+          begin
+            response = http_get(url_for("/.well-known/terraform.json"))
+            response.status == 200 ? JSON.parse(response.body) : {}
+          end
+      end
+
+      def service_key_for(name)
+        case name
```

Comment:
```suggestion
      def service_key_for(type)
        case type
```

Uber-nit: since this is the dependency's first requirement's source's type, `name` drops a lot of context.
Maybe `source_type`, `dep_type`, etc?